akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ReplicatedEventSourcingSpec.scala

```diff
@@ -211,18 +213,17 @@ class ReplicatedEventSourcingSpec
       val probe = createTestProbe[Done]()
       val eventProbeR1 = createTestProbe[EventAndContext]()
 
-      val latch1 = new CountDownLatch(1)
-      val latch2 = new CountDownLatch(1)
+      val latch = new CountDownLatch(3)
 
       val r1 = spawn(testBehavior(entityId, "R1", eventProbeR1.ref))
       val r2 = spawn(testBehavior(entityId, "R2"))
-      r1 ! StoreUs("1 from r1" :: "2 from r1" :: Nil, probe.ref, latch1)
-      r2 ! StoreUs("1 from r2" :: "2 from r2" :: Nil, probe.ref, latch2)
+      r1 ! StoreUs("1 from r1" :: "2 from r1" :: Nil, probe.ref, latch)
+      r2 ! StoreUs("1 from r2" :: "2 from r2" :: Nil, probe.ref, latch)
 
       // the commands have arrived in both actors, waiting for the latch,
       // so that the persist of the events will be concurrent
-      latch1.countDown()
-      latch2.countDown()
+      latch.countDown()
+      latch.await(10, TimeUnit.SECONDS)
```

Comment:
My guess what was going on: we didn't actually wait for the commands to arrive in each actor before counting down the latch, changed to be a single latch for all three threads - the two actors and this thread before continuing (even though this thread is not strictly necessary I thought it might make things more clear if all three continue at the "same time")